src/renderer/contextMenu/editor/spellcheck.js

```diff
@@ -0,0 +1,97 @@
+
+/**
+ * Build the spell checker menu depending on input.
+ *
+ * @param {[SpellChecker]} spellchecker The spellcheck wrapper.
+ * @param {[string]} selectedWord The selected word.
+ * @param {[string[]]} wordSuggestions Suggestions for `word`.
+ * @param {*} replaceCallback The callback to replace the word by a replacement.
+ * @returns {MenuItem[]}
+ */
+export default (spellchecker, selectedWord, wordSuggestions, replaceCallback) => {
+  if (spellchecker && spellchecker.isEnabled) {
+    const spellingSubmenu = []
+
+    // Change language menu entries
+    const currentLanguage = spellchecker.lang
+    const availableDictionaries = spellchecker.getAvailableDictionaries()
+    const availableDictionariesSubmenu = []
+    for (const dict of availableDictionaries) {
+      availableDictionariesSubmenu.push(new MenuItem({
+        label: dict,
+        enabled: dict !== currentLanguage,
+        click () {
+          bus.$emit('switch-spellchecker-language', dict)
+        }
+      }))
+    }
+
+    spellingSubmenu.push(new MenuItem({
+      label: 'Change Language...',
+      submenu: availableDictionariesSubmenu
+    }))
+
+    // TODO(spell): Delete me
+    spellingSubmenu.push(new MenuItem({
+      label: 'Debug',
+      click (menuItem, browserWindow) {
+        console.log('#########################')
+        console.log(spellchecker.getConfiguration)
+      }
+    }))
+
+    spellingSubmenu.push(SEPARATOR)
+
+    // Word suggestions
+    if (selectedWord && wordSuggestions && wordSuggestions.length > 0) {
+      spellingSubmenu.push({
+        label: 'Add to Dictionary',
+        click (menuItem, targetWindow) {
+          // NOTE: Need to notify Chromium to invalidate the spelling underline.
+          targetWindow.webContents.replaceMisspelling(selectedWord)
+          spellchecker.addToDictionary(selectedWord)
+            .catch(error => {
+              log.error(`Error while adding "${selectedWord}" to dictionary.`)
+              log.error(error)
+            })
+        }
+      })
+      spellingSubmenu.push(SEPARATOR)
+      for (const word of wordSuggestions) {
+        spellingSubmenu.push({
+          label: word,
+          click () {
+            console.log(`We'll replace "${word}".`) // // TODO(spell): Delete me
+
+            // Notify Muya to replace the word. We cannot just use Chromium to
+            // replace the word because the change is not forwarded to Muya.
+            replaceCallback(word)
+          }
+        })
+      }
+    } else {
+      spellingSubmenu.push({
+        label: 'Remove from Dictionary',
+        // NOTE: We cannot validate that the word is inside the user dictionary.
+        enabled: !!selectedWord && selectedWord.length >= 2,
+        click (menuItem, targetWindow) {
+          // NOTE: Need to notify Chromium to invalidate the spelling underline.
+          targetWindow.webContents.replaceMisspelling(selectedWord)
+          spellchecker.removeFromDictionary(selectedWord)
```

Comment:
What's the meaning of `removeFromDictionary`? will the `word` no longer be spell checked in other file?

Reply:
Yeah, I need to write user documentation for this feature. It removes a word that was added by the user to the spellchecker. The problem is that we cannot know whether the selected word is inside the user dictionary on macOS, so it's always enabled but only removes words that was added by the user.